sdk/java/dagger-java-annotation-processor/src/main/java/io/dagger/annotation/processor/DaggerModuleAnnotationProcessor.java

```diff
@@ -239,6 +184,76 @@ ModuleInfo generateModuleInfo(Set<? extends TypeElement> annotations, RoundEnvir
         moduleDescription, annotatedObjects.toArray(new ObjectInfo[annotatedObjects.size()]));
   }
 
+  private List<ParameterInfo> parseParameters(ExecutableElement elt) {
+    return elt.getParameters().stream()
+        .map(
+            param -> {
+              TypeMirror tm = param.asType();
+              TypeKind tk = tm.getKind();
+
+              boolean isOptional = false;
```

Comment:
I'm not a big fan of string manipulation when comparing types. Maybe this could be more robust:
```java
if (tm instanceof DeclaredType dt) {
  isOptional = dt.asElement().getQualifiedName().contentEquals(Optional.class.getCanonicalName());
  if (isOptional) {
    tm = dt.getTypeArguments().get(0);
  }
}
```

Reply:
This type system drives me crazy 🤪
But I finally have something working. Maybe not the most efficient, but working as expected.

```java
var optionalType =
    processingEnv.getElementUtils().getTypeElement(Optional.class.getName()).asType();
if (tm instanceof DeclaredType dt) {
  if (processingEnv.getTypeUtils().isSameType(dt.asElement().asType(), optionalType)) {
    isOptional = true;
    tm = dt.getTypeArguments().get(0);
    tk = tm.getKind();
  }
}
```

(will be push soon, once the base branch passes)